diff --git a/types/scene/surface.c b/types/scene/surface.c
index 1fe0762b..de30d280 100644
--- a/types/scene/surface.c
+++ b/types/scene/surface.c
@@ -110,7 +110,6 @@ static void set_buffer_with_surface_state(struct wlr_scene_buffer *scene_buffer,
 	wlr_surface_get_buffer_source_box(surface, &src_box);
 	wlr_scene_buffer_set_source_box(scene_buffer, &src_box);
 
-	wlr_scene_buffer_set_dest_size(scene_buffer, state->width, state->height);
 	wlr_scene_buffer_set_transform(scene_buffer, state->transform);
 
 	scene_buffer_unmark_client_buffer(scene_buffer);
