diff --git a/src/wl_window.c b/src/wl_window.c
index 5b491ffb..0fb95004 100644
--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -2227,8 +2227,8 @@ void _glfwSetWindowTitleWayland(_GLFWwindow* window, const char* title)
 void _glfwSetWindowIconWayland(_GLFWwindow* window,
                                int count, const GLFWimage* images)
 {
-    _glfwInputError(GLFW_FEATURE_UNAVAILABLE,
-                    "Wayland: The platform does not support setting the window icon");
+    /* _glfwInputError(GLFW_FEATURE_UNAVAILABLE, */
+    /*                 "Wayland: The platform does not support setting the window icon"); */
 }
 
 void _glfwGetWindowPosWayland(_GLFWwindow* window, int* xpos, int* ypos)
@@ -2236,8 +2236,8 @@ void _glfwGetWindowPosWayland(_GLFWwindow* window, int* xpos, int* ypos)
     // A Wayland client is not aware of its position, so just warn and leave it
     // as (0, 0)
 
-    _glfwInputError(GLFW_FEATURE_UNAVAILABLE,
-                    "Wayland: The platform does not provide the window position");
+    /* _glfwInputError(GLFW_FEATURE_UNAVAILABLE, */
+    /*                 "Wayland: The platform does not provide the window position"); */
 }
 
 void _glfwSetWindowPosWayland(_GLFWwindow* window, int xpos, int ypos)
@@ -2667,8 +2667,8 @@ void _glfwGetCursorPosWayland(_GLFWwindow* window, double* xpos, double* ypos)
 
 void _glfwSetCursorPosWayland(_GLFWwindow* window, double x, double y)
 {
-    _glfwInputError(GLFW_FEATURE_UNAVAILABLE,
-                    "Wayland: The platform does not support setting the cursor position");
+    /* _glfwInputError(GLFW_FEATURE_UNAVAILABLE, */
+    /*                 "Wayland: The platform does not support setting the cursor position"); */
 }
 
 void _glfwSetCursorModeWayland(_GLFWwindow* window, int mode)
